hooks/gen_docs.py

```diff
@@ -1,10 +1,263 @@
 """Generates the whole 'generatable' KPOps documentation"""
+import os
 import subprocess
+from pathlib import Path
+
+import yaml
 
 from hooks import PATH_ROOT
+from kpops.cli.registry import _find_classes
+from kpops.components.base_components.pipeline_component import PipelineComponent
+from kpops.utils.yaml_loading import load_yaml_file
 
 PATH_KPOPS_MAIN = PATH_ROOT / "kpops/cli/main.py"
 PATH_CLI_COMMANDS_DOC = PATH_ROOT / "docs/docs/user/references/cli-commands.md"
+PATH_DOCS_RESOURCES = PATH_ROOT / "docs/docs/resources"
+PATH_DOCS_COMPONENTS = PATH_DOCS_RESOURCES / "pipeline-components"
+PATH_DOCS_COMPONENTS_DEPENDENCIES = (
+    PATH_DOCS_COMPONENTS / "dependencies/pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS = (
+    PATH_DOCS_COMPONENTS / "dependencies/defaults_pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_KPOPS_STRUCTURE = PATH_DOCS_COMPONENTS / "dependencies/kpops_structure.yaml"
+KPOPS_COMPONENTS = tuple(_find_classes("kpops.components", PipelineComponent))
+KPOPS_COMPONENTS_INHERITANCE_REF: dict = {
+    component.get_component_type(): component.__base__.get_component_type()  # type: ignore[reportGeneralTypeIssues]
+    for component in KPOPS_COMPONENTS
+}
+KPOPS_COMPONENTS_FIELDS = {
+    # `set` to make it pickleable
+    component.get_component_type(): set(component.__fields__.keys())
+    for component in KPOPS_COMPONENTS
+}
+SECTIONS_ORDER = [
+    "type",
+    "name",
+    "namespace",
+    "app",
+    "from_",
+    "to",
+    "prefix",
+    "repo_config",
+    "version",
+    "resetter_values",
+    "offset_topic",
+]
+
+#####################
+# EXAMPLES          #
+#####################
+
+
+def filter_sections(
+    component_name: str, sections: list[str], include_inherited: bool = False
+) -> list[str]:
+    """Find all sections that are specific to a component from a list
+
+    :param component_name: Component name
+    :param sections: Available section files, names with extension, no path
+    :param include_inherited: Whether to include sections that can be defined in
+        a parent component defaults
+    :returns: The applicable sections
+    """
+    component_sections: list[str] = []
+    for target_section in KPOPS_COMPONENTS_FIELDS[component_name]:
+        component = component_name
```

Comment:
why not leave it as `component_name` if it's just a str, not the actual component class?

Reply:
I just needed a new name to do [this](https://github.com/bakdata/kpops/pull/267/files/d127dec5a420ff87cf3f3c9f0934773ab7519b87#diff-69ba40347f1f5ed3ff5374d0a0252fed381b703ca07b32ecd11089956c6ff4a5R79)

Reply:
sorry, I still don't get it. If I remove this line and rename later uses of `component` to `component_name` it seems to work as well

Reply:
The idea is that in [this branch](https://github.com/bakdata/kpops/blob/ci/concatenate-examples/hooks/gen_docs.py#L78-L88) I reassign the component name to the parent while it is not `PipelineComponent` in search for the relevant section to inherit. I was worried that if I don't use a "burner" variable, the function would no longer be solving for the same component in further iterations.

Reply:
If I assign a different value to `component_name`, would it not remain change for the rest of the function's life?

Reply:
but at this point the dict access in the surrounding for loop is already evaluated, no?

Reply:
[Here](https://github.com/bakdata/kpops/blob/ci/concatenate-examples/hooks/gen_docs.py#L67C25-L67C25) I use it to assign a name to a file

Reply:
I think I may have some general misunderstanding how variables behave in such loops haha. The whole idea is to reset the component name to its original value for each iteration.

Reply:
@disrupted Let me know if you still see a problem, I would be happy to simplify the function :)

Else, please mark as resolved